services/py-api/src/service/mail_service/resend_service.py

```diff
@@ -0,0 +1,95 @@
+import os
+from typing import Optional
+from dotenv import load_dotenv
+
+import resend
+from mail_service import MailService
+from service.mail_service.utils import load_email_participant_html_template, load_email_verify_participant_html_template
+
+SENDER_EMAIL = "tonymontana@thehub.com"  # We obviously need to supply our own email here.
+load_dotenv()
+
+
+class ResendMailService(MailService):
+    def __init__(self) -> None:
+        api_key = os.getenv("RESEND_API_KEY")
+        if not api_key:
+            raise ValueError("RESEND_API_KEY environment variable is not set")
+        resend.api_key = api_key
+        self.client = resend
+
+    async def send_email(
+        self, receiver: str, subject: str, body_text: Optional[str] = None, body_html: Optional[str] = None
+    ) -> None:
+        if not receiver:
+            raise ValueError("Receiver email must be provided.")
+        if not subject:
+            raise ValueError("Subject must be provided.")
+
+        if not body_text and not body_html:
+            raise ValueError("At least one of body_text or body_html must be provided")
+
+        params: resend.Emails.SendParams = {
+            "from": f"THE HUB <{SENDER_EMAIL}>",
+            "to": [receiver],
+            "subject": subject,
+        }
+
+        if body_html:
+            params["html"] = body_html
+        elif body_text:
+            params["text"] = body_text
+
+        try:
+            email = self.client.Emails.send(params)
+            # TODO: I need to actually log here
+            print("Email sent successfully: ", email)
+        except Exception as e:
+            raise RuntimeError(f"Failed to send email: {str(e)}") from e
```

Comment:
We use the [Result pattern](https://www.milanjovanovic.tech/blog/functional-error-handling-in-dotnet-with-the-result-pattern) via [this](https://github.com/rustedpy/result) across our code base, so I suggest you to do: 
`return Err(...)` and log the exception with `LOG.exception`. Here is a good example:
https://github.com/AUBGTheHUB/monolith/blob/30fe96921c1b8f212274389057d5bc39b5aed654/services/py-api/src/database/transaction_manager.py#L143-L146

otherwise, as this method with your code version raises exceptions, you should declare this in the docs so that the caller knows that it has to do some try except action. But the caller should read the docs :D, using `Result` forces you to think more about error handling, as you know by the method definition that it could return an `Err`.  As mentioned in the article "Exceptions are for exceptional situations."

Reply:
What a exceptional explanation by @IvanObreshkov! ⭐️